File: Java - Coding Challenges/chessKnight.java
/**
 * Given a position of a knight on the standard chessboard, find the number of different moves the knight can perform.
 * 
 * Example:
 * For cell = "a1", the output should be
 * solution(cell) = 2.
 *
 * [execution time limit] 3 seconds (java)
 * [input] string cell
 * 
 * [output] integer
 * 
 * Tests passed: 20/20.
 */
int solution(String cell) {
    String[] letters = {"a","b","c","d","e","f","g","h"};
    int count = 0;
    
    // Vertical Up Right.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+1 >= 0 && Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+1 <= letters.length-1) && (Character.getNumericValue(cell.charAt(1))+2 <= 8) ) {
        System.out.println("1");
        count++;
    }
    
    // Vertical Up Left.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))-1 >= 0) && (Character.getNumericValue(cell.charAt(1))+2 <= 8) ) {
        System.out.println("2");
        count++;
    }
    
     // Vertical Down Right.
     if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+1 >= 0 && Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+1 <= letters.length-1) && (Character.getNumericValue(cell.charAt(1))-2 >= 1)) {
        System.out.println("3");
        count++;
    }
    
    // Vertical Down Left.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))-1 >= 0) && (Character.getNumericValue(cell.charAt(1))-2 >= 1) ) {
        System.out.println("4");
        count++;
    }
    
    // Horizontal Right Up.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+2 >= 0 && Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+2 <= letters.length-1) && (Character.getNumericValue(cell.charAt(1))+1 <= 8)) {
        System.out.println("5");
        count++;
    }

    // Horizontal Right Down.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+2 >= 0 && Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))+2 <= letters.length-1) && (Character.getNumericValue(cell.charAt(1))-1 >= 1)) {
        System.out.println("6");
        count++;
    }
    
    // Horizontal Left Up.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))-2 >= 0) && (Character.getNumericValue(cell.charAt(1))+1 >= 1 && Character.getNumericValue(cell.charAt(1))+1 <= 8) ) {
        System.out.println("7");
        count++;
    }
    
    // Horizontal Left Down.
    if ((Arrays.asList(letters).indexOf(String.valueOf(cell.charAt(0)))-2 >= 0) && (Character.getNumericValue(cell.charAt(1))-1 >= 1)) {
        System.out.println("8");
        count++;
    }
    
    return count;
}

File: JavaScript - Coding Challenges/chess-knight.js
/**
 * Given a position of a knight on the standard chessboard,
 * find the number of different moves the knight can perform. 
 * 
 * [execution time limit] 4 seconds (js)
 * [input] string cell
 * 
 * [output] integer
 * 
 * Tests passed: 20/20.
 */
function solution(cell) {
    const letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h'];
    const numbers = [1, 2, 3, 4, 5, 6, 7, 8];
    let   count = 0;
    // vertical up right.
    if ((0 <= letters.indexOf(cell[0]) + 1 && letters.indexOf(cell[0]) + 1 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) + 2))) {
        count++;
    }
    // vertical up left.
    if ((0 <= letters.indexOf(cell[0]) - 1 && letters.indexOf(cell[0]) - 1 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) + 2))) {
        count++;
    }
    // vertical down right.
    if ((0 <= letters.indexOf(cell[0]) + 1 && letters.indexOf(cell[0]) + 1 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) - 2))) {
        count++;
    }
    // vertical down left.
    if ((0 <= letters.indexOf(cell[0]) - 1 && letters.indexOf(cell[0]) - 1 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) - 2))) {
        count++;
    }
    // horizontal right up.
    if ((0 <= letters.indexOf(cell[0]) + 2 && letters.indexOf(cell[0]) + 2 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) + 1))) {
        count++;
    }
    // horizontal right down.
    if ((0 <= letters.indexOf(cell[0]) + 2 && letters.indexOf(cell[0]) + 2 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) - 1))) {
        count++;
    }
    // horizontal left up.
    if ((0 <= letters.indexOf(cell[0]) - 2 && letters.indexOf(cell[0]) - 2 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) + 1))) {
        count++;
    }
    // horizontal left down.
    if ((0 <= letters.indexOf(cell[0]) - 2 && letters.indexOf(cell[0]) - 2 <= letters.length - 1) && (numbers.includes(parseInt(cell[1]) - 1))) {
        count++;
    }
    return count;
}